kafka/metrics_test.go

```diff
@@ -322,20 +322,22 @@ func setupTestProducer(t testing.TB) (*Producer, sdkmetric.Reader) {
 
 	rdr := sdkmetric.NewManualReader()
 	_, brokers := newClusterWithTopics(t, 1, "name_space-default-topic")
-	producer, err := NewProducer(ProducerConfig{
+	mp := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(rdr),
+	)
+	t.Cleanup(func() {
+		require.NoError(t, mp.Shutdown(context.Background()))
```

Comment:
What's the reason for checking the error for this MeterProvider and not for the TracerProvider [here](https://github.com/elastic/apm-queue/pull/264/files#diff-39d391f5443633a22bf7611a2ce05ed77678f040ccf56a5d681a3e60f317d291R111)?